### PR TITLE
add suffix on id when copying load

### PR DESCRIPTION
**Pull request by AbdelHedhili**

### CODE REVIEW

File: src/utils/rest-api.js [hunk 1]
```diff
@@ -484,6 +484,25 @@ function fetchEquipmentInfos(
     return backendFetch(fetchEquipmentInfosUrl);
 }
 
+export function fetchEquipmentExists(
```
Comment (SlimaneAmar) [inline src/utils/rest-api.js:487]: Need to add a head method  to avoid to get all equipment infos at each iteration ?
Need just to know if it exists

**Review by SlimaneAmar — CHANGES_REQUESTED** [review]

**Review by sBouzols — APPROVED** [review]

File: src/utils/rest-api.js (hunk 1 above)
Comment (AbdelHedhili) [reply src/utils/rest-api.js:487]: it already exists that's what I used

Comment (AbdelHedhili) [reply src/utils/rest-api.js:487]: there is fetchEquipment and fetchEquipmentS

File: src/components/dialogs/load-creation-dialog.js [hunk 2]
```diff
@@ -216,35 +220,52 @@ const LoadCreationDialog = ({
...
-                                loadId: element.id,
-                            }
-                        );
-                        enqueueSnackbar(msg, {
-                            variant: 'info',
-                            persist: false,
-                            style: { whiteSpace: 'pre-line' },
+                        addSuffixIfNecessary(load.id, 1).then((loadId) => {
+                            setLoadId(loadId);
+                            //For now we don't want to retrieve nor try to set the BusBarSection, users have to select it.
```
Comment (sBouzols) [inline src/components/dialogs/load-creation-dialog.js:246]: maybe you could move this comment near 
`setBusOrBusbarSection(null);`
which is the line concern by this comment

File: src/utils/rest-api.js [hunk 3]
```diff
@@ -469,21 +476,52 @@ function fetchEquipmentInfos(
...
+    studyUuid,
+    selectedNodeUuid,
+    equipmentPath,
+    equipmentId
+) {
+    console.info(
+        `Fetching existence of specific equipments '${equipmentId}' of type '${equipmentPath}' of study '${studyUuid}' and node '${selectedNodeUuid}' ...`
+    );
+    const fetchEquipmentInfosUrl =
```
Comment (sBouzols) [inline src/utils/rest-api.js:515]: rename `fetchEquipmentInfosUrl` to `fetchEquipmentExistsUrl` to avoid confusion

File: src/utils/rest-api.js [hunk 4]
```diff
@@ -469,21 +476,52 @@ function fetchEquipmentInfos(
...
+        urlSearchParams.toString();
     console.debug(fetchEquipmentInfosUrl);
     return backendFetch(fetchEquipmentInfosUrl);
 }
 
+export function fetchEquipmentExists(
+    studyUuid,
+    selectedNodeUuid,
+    equipmentPath,
+    equipmentId
```
Comment (sBouzols) [inline src/utils/rest-api.js:510]: use `searchInUpstreamBuiltParent` as well here ?

**Review by sBouzols — CHANGES_REQUESTED** [review]
Code review
No tests

File: src/components/dialogs/load-creation-dialog.js (hunk 2 above)
Comment (AbdelHedhili) [reply src/components/dialogs/load-creation-dialog.js:247]: done

File: src/utils/rest-api.js (hunk 3 above)
Comment (AbdelHedhili) [reply src/utils/rest-api.js:523]: done

File: src/utils/rest-api.js (hunk 4 above)
Comment (AbdelHedhili) [reply src/utils/rest-api.js]: fixed

**Review by sBouzols — APPROVED** [review]
Code Review OK 
Test OK 
- create load from another
- create load from copy
- create load from an hypo node under another load creation to check if it takes effeect in search
- test if fields are correctly overwritten
- test if busbarsection list is updated

File: src/components/dialogs/load-creation-dialog.js [hunk 5]
```diff
@@ -216,35 +220,53 @@ const LoadCreationDialog = ({
         setDialogSearchOpen(true);
     };
 
+    const addSuffixIfNecessary = (loadId, i) => {
```
Comment (SlimaneAmar) [inline src/components/dialogs/load-creation-dialog.js:223]: Remove it or do it on the back side

**Review by SlimaneAmar — CHANGES_REQUESTED** [review]

File: src/components/dialogs/load-creation-dialog.js (hunk 5 above)
Comment (AbdelHedhili) [reply src/components/dialogs/load-creation-dialog.js]: removed

File: src/utils/rest-api.js [hunk 6]
```diff
@@ -412,6 +413,12 @@ export function fetchEquipmentsInfos(
     let urlSearchParams = new URLSearchParams();
     urlSearchParams.append('userInput', searchTerm);
     urlSearchParams.append('fieldSelector', usesName ? 'name' : 'id');
+    if (searchInUpstreamBuiltParent !== undefined) {
```
Comment (SlimaneAmar) [inline src/utils/rest-api.js:416]: !searchInUpstreamBuiltParent ?

File: src/utils/rest-api.js [hunk 7]
```diff
@@ -469,17 +482,29 @@ function fetchEquipmentInfos(
...
-    equipmentId
+    equipmentId,
+    searchInUpstreamBuiltParent
 ) {
     console.info(
         `Fetching specific equipments '${equipmentId}' of type '${equipmentPath}' of study '${studyUuid}' and node '${selectedNodeUuid}' ...`
     );
+
+    let urlSearchParams = new URLSearchParams();
+    if (searchInUpstreamBuiltParent !== undefined) {
```
Comment (SlimaneAmar) [inline src/utils/rest-api.js:493]: !searchInUpstreamBuiltParent ?

**Review by SlimaneAmar — APPROVED** [review]

**Review by FranckLecuyer — APPROVED** [review]

**Review by SlimaneAmar — APPROVED** [review]

**Review by JacquesBorsenberger — APPROVED** [review]

**Pull request merged**